### ISSUE TITLE
Removed rules not listed in v5->v6 migration guide

### ISSUE THREAD

**Issue by connorpwilliams** [stoplightio/spectral#1765]
**Chore summary**
Add/update missing rules
The v5 to v6 migration guide did not have any documentation as to what happened with these three rules. Currently they are identified as non-existent.
Guide needs to be updated to indicated if the rules were merged with something else or removed.

**Tasks**
- [ ] model-decription rule - non existent rule (removed?)
- [ ] example-value-or-externalValue - non existent rule (removed?)
- [ ] operation-summary-formatted - non existent rule (removed?)

**Additional context**
For the time being I have just removed them from my ruleset to workaround the "Invalid spectral file" issue.
Not high priority as I have a workaround, but if these have been merged I'd like to replace them instead of remove them.

Thanks!
Connor
<br><br><i>(related to <a href='https://stoplight.zendesk.com/agent/tickets/5498'>Zendesk ticket #5498</a>)<br>gz#5498</i>

**Comment by P0lip** [comment]
Hey!
Thanks for raising the issue.
All the rules listed don't exist already in the latest v5, thus they aren't listed in the v6 migration guide.

- `example-value-or-externalValue` has been renamed to `oas3-examples-value-or-externalValue`.

- `operation-summary-formatted` has been removed and we have no equivalent available. If you're still interested in that rule, I suggest to add it to your ruleset.
Here's the older definition you can use in your own ruleset:
   ```json
   "operation-summary-formatted": {
      "description": "Operation `summary` should start with upper case and end with a dot.",
      "given": "$.paths[*][get,put,post,delete,options,head,patch,trace",
      "then": {
        "field": "summary",
        "function": "pattern",
        "functionOptions": {
          "match": "^[A-Z].*\\.$"
        }
      }
    }
  ```

- `model-description` - re this one, I'm not sure whether this one ever existed. W have `tag-description`, however, so perhaps that's what was meant?

Let me know if you have any questions or if something needs further clarification.
Happy to help.

**Comment by connorpwilliams** [comment]
Ah! I didn't know they were removed until I was getting errors this week about "You spectral file is invalid" and when opening the file I saw that the rules were non-existent.

Thank you for this, I will update my file accordingly.

Is there a way you could consider reporting rules as deprecated before removing them? This would cause a lot of my projects to not have broken rulesets and provide an easy way for me to know when rules are going to change/be removed.
Just some feedback! 😀

**Comment by P0lip** [comment]
> Is there a way you could consider reporting rules as deprecated before removing them? This would cause a lot of my projects to not have broken rulesets and provide an easy way for me to know when rules are going to change/be removed.

Yup, totally. That's a very good idea. We do leave deprecation warnings usually in the code, but never in rulesets, so something for us to look into.

> You spectral file is invalid

Ah, my apologies, that's presumably in Studio?
Spectral has a much better ruleset validation as of the newer version, both in terms of strictness (some user "mistakes" are considered actual errors now like these non-existent rules) & improved error messages, but not all parts of this are hooked up in Studio just yet.

In any case, do feel free to reach out to us, always happy to help. Once again apologies for any inconvenience caused.

---

By the way, with that aforementioned new release of Spectral, one is able to stick with a particular version of a ruleset, so it's possible to remain immune to further changes we make if one wants is satisfied with their current setup.
In other words, you can choose the version of the OAS ruleset you want and forget about it. In such a scenario, even if we introduce breaking changes to the rulesets, they won't be automatically propagated to your projects / rulesets unless you explicitly change the version of the ruleset.
We don't have much documentation around it just yet, but if that's something interesting for you, I'll make sure to follow up once we have a more concrete tutorial published.
